src/python/pants/backend/docker/goals/package_image_test.py

```diff
@@ -131,16 +136,108 @@ def run_process_mock(process: Process) -> FallibleProcessResult:
         docker_options = rule_runner.request(DockerOptions, [])
 
     global_options = rule_runner.request(GlobalOptions, [])
-
+    build_request = BuildDockerImageRequest(
+        context=build_context_mock(),
+        field_set=DockerFieldSet.create(tgt),
+        target=tgt,
+    )
     result = run_rule_with_mocks(
         build_docker_image,
         rule_args=[
-            DockerFieldSet.create(tgt),
+            build_request,
             docker_options,
             global_options,
             DockerBinary("/dummy/docker"),
             ProcessCleanupOption(True),
         ],
+        mock_gets=[
+            MockGet(
+                output_type=FallibleProcessResult,
+                input_type=Process,
+                mock=run_process_mock,
+            ),
+        ],
+    )
+
+    # Package up result as done in the `@rule package_image()` now..
+    built = BuiltPackage(
+        result.digest,
+        (BuiltDockerImage.create(result.tags),),
+    )
+
+    assert built.digest == EMPTY_DIGEST
+    assert len(built.artifacts) == 1
+    assert built.artifacts[0].relpath is None
+
+    for log_line in extra_log_lines:
+        assert log_line in built.artifacts[0].extra_log_lines
+
+
+def assert_package(
```

Comment:
Leaving this for cleanup in a follow up PR, to unify and parametrize fixtures to reduce duplication and test bloat for these two assert functions (assert_build and assert_package).